### ISSUE TITLE
[macOS] Nintendo Switch Pro Controller is non-functional over USB

### ISSUE THREAD

**Issue by DJtheMan2101** [libretro/RetroArch#16136]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

When an official Nintendo Switch Pro Controller is connected via USB, none of its inputs function in RetroArch at all.

### Expected behavior

The controller should work so it can navigate RetroArch's menus and control the cores. The controller does work over USB in other programs (ES-DE, Dolphin, and PCSX2). The controller _does_ work in RetroArch via Bluetooth and the "mfi" controller driver, but macOS Sonoma has a bug that causes it to have significant input lag over Bluetooth, so USB is preferable.

### Actual behavior

When the controller is connected via USB, [its autoconfig profile](https://github.com/libretro/retroarch-joypad-autoconfig/blob/master/hid/Pro%20Controller.cfg) is loaded (assuming the "hid" controller driver is set), but nothing else happens. Whether the profile is present or not, none of the inputs will do anything in RetroArch's menu or cores. In previous nightly builds, the controller would sometimes instead send random inputs on every frame, but I cannot recreate this behavior reliably.

The controller also does not work via USB with the "mfi" controller driver, but I assume that is expected (the profile is not loaded in that case). I do not have any other controllers to test, so I do not know if the issue is specific to the Nintendo Switch Pro Controller or if this is a problem with the "hid" controller driver and/or the macOS version of RetroArch in general.

### Steps to reproduce the bug

1. Open RetroArch on a Mac with the controller driver set to "hid".
2. Connect a Nintendo Switch Pro Controller to your Mac via USB.
3. See that the autoconfig is loaded, but the controller does nothing.

### Bisect Results

In stable 1.9.0 (the earliest available stable build), the controller's (non-)behavior seems to be the same as in the latest nightly. I usually use the controller via Bluetooth myself, so I have no idea if it ever worked via USB on macOS.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.16.0 (nightly 2024-01-18)

### Environment information

- OS: macOS Sonoma 14.2.1

**Comment by erfg12** [comment]
RetroArch 1.17.0 (stable Metal)

Still not working...

**Comment by sarahckohl** [comment]
Hello,
I was able to solve the issue of the controller not registering while plugged in via usb-c by manually toggling off game mode.  Game mode sometimes interferes with background processes that ironically are needed to play games.  

Furthermore there is a setting in RetroArch which automatically toggles Game Mode on, you will want to turn it off:
Settings > Input > Auto Enable 'Game Mode'.

I also found @erfg12 's suggestion regarding changing Analog to Digital type to None.  I had previously set it to left stick thinking this was necessary to have analog support on the Pro Controller.  It's not.  Instead, go to the Quick Menu while running Swanstation, go to Port 1 and change the device type from Digital Controller to Analog (DualShock).

Finally, I followed  @erfg12 and changed the input type from Digital Controller to Analog (DualShock), and set Analog to Digital Type to None which I had previously set to Left stick thinking that was the only wa
